### PR TITLE
Fix triton_fused_moe unit test and benchmark

### PR DESCRIPTION
**Pull request by yuan-luo**
## Motivation

The test_triton_fused_moe and benchmark_sglang_fused_moe_triton have been broken due to recent refactor. This PR is to fix these cases.

**Unit test result:**
```
$python ./test/srt/test_triton_fused_moe.py
INFO 08-17 21:26:58 [__init__.py:235] Automatically detected platform cuda.
All deep_gemm operations loaded successfully!
WARNING:sglang.srt.layers.quantization.deep_gemm_wrapper.compile_utils:NVCC Compiler not found, use NVRTC for DeepGEMM JIT and may have performance loss with some cases.
INFO 08-17 21:26:59 [__init__.py:235] Automatically detected platform cuda.
INFO 08-17 21:26:59 [__init__.py:235] Automatically detected platform cuda.
INFO 08-17 21:26:59 [__init__.py:235] Automatically detected platform cuda.
INFO 08-17 21:26:59 [__init__.py:235] Automatically detected platform cuda.
INFO 08-17 21:26:59 [__init__.py:235] Automatically detected platform cuda.
W0817 21:26:59.966000 46222 site-packages/torch/utils/cpp_extension.py:2425] TORCH_CUDA_ARCH_LIST is not set, all archs for visible cards are included for compilation. 
W0817 21:26:59.966000 46222 site-packages/torch/utils/cpp_extension.py:2425] If this is not desired, please set os.environ['TORCH_CUDA_ARCH_LIST'] to specific architectures.
WARNING:sglang.srt.utils:MOE_RUNNER_BACKEND is not initialized, using triton backend
Running MoE tests: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 96/96 [00:02<00:00, 34.13it/s]
.
----------------------------------------------------------------------
Ran 1 test in 2.814s

OK
```

**Benchmark triton kernel with enable cuda graph**
```
$python ./benchmark/kernels/fused_moe_triton/benchmark_sglang_fused_moe_triton.py --use-cuda-graph
INFO 08-18 17:18:53 [__init__.py:235] Automatically detected platform cuda.
All deep_gemm operations loaded successfully!
WARNING:sglang.srt.layers.quantization.deep_gemm_wrapper.compile_utils:NVCC Compiler not found, use NVRTC for DeepGEMM JIT and may have performance loss with some cases.
INFO 08-18 17:18:54 [__init__.py:235] Automatically detected platform cuda.
INFO 08-18 17:18:54 [__init__.py:235] Automatically detected platform cuda.
INFO 08-18 17:18:54 [__init__.py:235] Automatically detected platform cuda.
INFO 08-18 17:18:54 [__init__.py:235] Automatically detected platform cuda.
INFO 08-18 17:18:54 [__init__.py:235] Automatically detected platform cuda.
W0818 17:18:54.698000 66458 site-packages/torch/utils/cpp_extension.py:2425] TORCH_CUDA_ARCH_LIST is not set, all archs for visible cards are included for compilation. 
W0818 17:18:54.698000 66458 site-packages/torch/utils/cpp_extension.py:2425] If this is not desired, please set os.environ['TORCH_CUDA_ARCH_LIST'] to specific architectures.
WARNING:sglang.srt.utils:MOE_RUNNER_BACKEND is not initialized, using triton backend
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
shape_configs={'num_experts': 128, 'topk': 8, 'hidden_size': 2048, 'shard_intermediate_size': 768, 'dtype': torch.bfloat16, 'block_shape': None}
benchmark sglang_fused_moe_triton_v340 with batch_size=128
benchmark sglang_fused_moe_triton with batch_size=128
WARNING:sglang.srt.layers.moe.fused_moe_triton.fused_moe:Config file not found at /opt/conda/lib/python3.10/site-packages/sglang/srt/layers/moe/fused_moe_triton/configs/triton_3_4_0/E=128,N=384,device_name=NVIDIA_H20.json. Fallback to triton version 3.2.0 and use MoE kernel config from /opt/conda/lib/python3.10/site-packages/sglang/srt/layers/moe/fused_moe_triton/configs/triton_3_2_0/E=128,N=384,device_name=NVIDIA_H20.json. Performance might be sub-optimal!
benchmark sglang_fused_moe_triton_v340 with batch_size=256
benchmark sglang_fused_moe_triton with batch_size=256
benchmark sglang_fused_moe_triton_v340 with batch_size=512
benchmark sglang_fused_moe_triton with batch_size=512
benchmark sglang_fused_moe_triton_v340 with batch_size=1024
benchmark sglang_fused_moe_triton with batch_size=1024
benchmark sglang_fused_moe_triton_v340 with batch_size=2048
benchmark sglang_fused_moe_triton with batch_size=2048
benchmark sglang_fused_moe_triton_v340 with batch_size=4096
benchmark sglang_fused_moe_triton with batch_size=4096
benchmark sglang_fused_moe_triton_v340 with batch_size=8192
benchmark sglang_fused_moe_triton with batch_size=8192
fused-moe-performance:
   batch_size  sglang_fused_moe_triton_v340  sglang_fused_moe_triton
0       128.0                      0.232000                 0.209408
1       256.0                      0.304416                 0.237200
2       512.0                      0.477056                 0.320576
3      1024.0                      0.518464                 0.479488
4      2048.0                      0.991968                 0.761376
5      4096.0                      1.614560                 1.384752
6      8192.0                      2.870144                 2.592288

```

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

File: test/srt/test_triton_fused_moe.py [hunk 1]
```diff
@@ -92,8 +94,22 @@ def _test_case(self, m, n, k, e, topk, dtype):
         w2_tri = w2_tri.transpose(-2, -1).contiguous()
         score = self.create_random_cuda_tensor((m, e), dtype)
 
+        topk_output = TopK(
+            top_k=topk,
+            renormalize=False,
+            use_grouped_topk=False,
+        )
+        topk_output.use_triton_kernels = True
```
[the remark this answers is not in the document]
Comment (ch-wan) [reply test/srt/test_triton_fused_moe.py]: I have the same comment

File: test/srt/test_triton_fused_moe.py [hunk 2]
```diff
@@ -92,8 +94,22 @@ def _test_case(self, m, n, k, e, topk, dtype):
...
+        triton_topk_output = topk_output.forward_cuda(
+            hidden_states=a,
+            router_logits=score,
+        )
+
+        moe_runner_config = MoeRunnerConfig(
+            inplace=True,
```
Comment (ch-wan) [inline test/srt/test_triton_fused_moe.py]:
```suggestion
            inplace=False,
```
I remember that triton kernel does not support inplace op.

File: test/srt/test_triton_fused_moe.py [hunk 3]
```diff
@@ -92,8 +94,22 @@ def _test_case(self, m, n, k, e, topk, dtype):
         w2_tri = w2_tri.transpose(-2, -1).contiguous()
         score = self.create_random_cuda_tensor((m, e), dtype)
 
+        topk_output = TopK(
```
Comment (ch-wan) [inline test/srt/test_triton_fused_moe.py]:
```suggestion
        topk = TopK(
```

File: benchmark/kernels/fused_moe_triton/benchmark_sglang_fused_moe_triton.py [hunk 4]
```diff
@@ -80,13 +82,26 @@ def fused_moe_triton_api(
     input_gating,
     topk,
 ):
+    topk_output = TopK(
```
Comment (ch-wan) [inline benchmark/kernels/fused_moe_triton/benchmark_sglang_fused_moe_triton.py]:
```suggestion
    topk = TopK(
```

File: benchmark/kernels/fused_moe_triton/benchmark_sglang_fused_moe_triton.py [hunk 5]
```diff
@@ -80,13 +82,26 @@ def fused_moe_triton_api(
...
+        top_k=topk,
+        renormalize=False,
+        use_grouped_topk=False,
+    )
+    topk_output.use_triton_kernels = True
```
Comment (ch-wan) [inline benchmark/kernels/fused_moe_triton/benchmark_sglang_fused_moe_triton.py]: why not set it in its init stage?

File: benchmark/kernels/fused_moe_triton/benchmark_sglang_fused_moe_triton.py [hunk 6]
```diff
@@ -103,20 +118,21 @@ def fused_moe_sglang_api(
...
-        topk,
-        renormalize=False,
-        inplace=True,
+        topk_output,
         use_fp8_w8a8=use_fp8_w8a8,
         w1_scale=w1_scale,
         w2_scale=w2_scale,
         a1_scale=a1_scale,
         a2_scale=a2_scale,
-        block_shape=block_shape,
```
Comment (ch-wan) [inline benchmark/kernels/fused_moe_triton/benchmark_sglang_fused_moe_triton.py]: why it is removed?

File: test/srt/test_triton_fused_moe.py (hunk 2 above)
Comment (yuan-luo) [reply test/srt/test_triton_fused_moe.py]: Revised.

File: benchmark/kernels/fused_moe_triton/benchmark_sglang_fused_moe_triton.py (hunk 5 above)
Comment (yuan-luo) [reply benchmark/kernels/fused_moe_triton/benchmark_sglang_fused_moe_triton.py]: This happens because in TopK.__init__() the line
```
self.use_triton_kernels = get_moe_runner_backend().is_triton_kernel()
```
checks the backend, but in our tests get_moe_runner_backend() fails to pick up MOE_RUNNER_BACKEND="triton".
Therefore we simply hard-code it here.

File: benchmark/kernels/fused_moe_triton/benchmark_sglang_fused_moe_triton.py (hunk 6 above)
Comment (yuan-luo) [reply benchmark/kernels/fused_moe_triton/benchmark_sglang_fused_moe_triton.py]: It's because block_shape has default value None. I added back.

File: benchmark/kernels/fused_moe_triton/benchmark_sglang_fused_moe_triton.py (hunk 4 above)
Comment (yuan-luo) [reply benchmark/kernels/fused_moe_triton/benchmark_sglang_fused_moe_triton.py]: Revised to topk_op.

File: test/srt/test_triton_fused_moe.py (hunk 3 above)
Comment (yuan-luo) [reply test/srt/test_triton_fused_moe.py]: Revised to topk_op.

**Review by ch-wan — APPROVED** [review]
